### PR TITLE
Feature/st 29/question

### PR DESCRIPTION
**Pull request by VictoriaNegrotto**
Changed to question. Finished validations, controllers, routes, models.

### CODE REVIEW

**Comment by LucianoStradiot** [comment]
[Products entity](https://trello.com/c/3eeZjERI/29-products-entity)

File: src/controllers/question.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
+import Questions from '../models/Question';
+import User from '../models/User';
+//import Bid from '../models/Bid';
```
Comment (Tilajc) [inline src/controllers/question.js:3]:
```suggestion

```

File: src/controllers/question.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    return res.status(500).json({
+      message: `An error ocurred: ${error}`,
+      data: undefined,
+      error: true
+    });
+  }
+};
+
+const getQuestionById = async (req, res) => {
+  try {
```
Comment (Tilajc) [inline src/controllers/question.js:28]:
```suggestion
  if(!mongoose.isValidObjectId){
     return res.....
  }
  try {
```

File: src/controllers/question.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    });
+    if (!idQuestion) {
+      return res.status(404).json({
+        message: `Question with ID ${id} was not found`,
+        data: undefined,
+        error: true
+      });
+    }
+    return res.status(200).json({
+      message: `Question with ID ${idQuestion.id} was found!`,
```
Comment (Tilajc) [inline src/controllers/question.js:43]: Here the ID is enough to show

```suggestion
      message: `Question with ID ${id} was found!`,
```

File: src/controllers/question.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        error: true
+      });
+    }
+    const newQuestion = await Questions.create({
+      comment: newComment,
+      bid: newBid,
+      user: newUser
+    });
+    return res.status(201).json({
+      message: `Question with ID ${newQuestion.id} created!`,
```
Comment (Tilajc) [inline src/controllers/question.js:76]:
```suggestion
      message: 'Question created succesfully',
```

File: src/controllers/question.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+      message: 'ID invalid',
+      data: undefined,
+      error: true
+    });
+  }
+  try {
+    const currentQuestion = await Questions.findById(id);
+    if (!currentQuestion) {
+      return res.status(404).json({
+        message: `Question with id: ${id} was not found`,
```
Comment (Tilajc) [inline src/controllers/question.js:107]:
```suggestion
        message: `Question with ID: ${id} was not found`,
```

File: src/controllers/question.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  try {
+    const { id } = req.params;
+    const idQuestion = await Questions.findOne({ $and: [{ _id: id }] }).populate('user', {
+      firstName: 1,
+      lastName: 1,
+      profilePhoto: 1
+    });
+    if (!idQuestion) {
+      return res.status(404).json({
+        message: `Question with ID ${id} was not found`,
```
Comment (Tilajc) [inline src/controllers/question.js:37]:
```suggestion
        message: `Question with ID: ${id} was not found`,
```

File: src/controllers/question.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: `Question with ID ${idQuestion.id} was found!`,
+      data: idQuestion,
+      error: false
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: error,
```
Comment (Tilajc) [inline src/controllers/question.js:49]:
```suggestion
      message: `An error ocurred: ${error}`,
```

File: src/controllers/question.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+      user: newUser
+    });
+    return res.status(201).json({
+      message: `Question with ID ${newQuestion.id} created!`,
+      data: newQuestion,
+      error: false
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: error,
```
Comment (Tilajc) [inline src/controllers/question.js:82]:
```suggestion
      message: `An error ocurred: ${error}`,
```

File: src/validations/question.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  const errorMessage = validationResult.error.details.map((details) => details.message);
+
+  return res.status(400).json({
+    message: `There was en error ${errorMessage.join(', ')}`,
+    data: undefined,
+    error: true
+  });
+};
+
+const validateupdateQuestion = (req, res, next) => {
```
Comment (Tilajc) [inline src/validations/question.js:30]:
```suggestion
const validateUpdateQuestion = (req, res, next) => {
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
Pretty good job, the job is almost done, just little comments for consistence!

**Review by LucianoStradiot — CHANGES_REQUESTED** [review]
With the tila´s correction It should be finished

**Review by LucianoStradiot — APPROVED** [review]
LG!

**Review by Tilajc — APPROVED** [review]

**Pull request merged**